includes/sanitizers/class-amp-tag-and-attribute-sanitizer.php

```diff
@@ -922,6 +923,25 @@ private function validate_cdata_for_node( DOMElement $element, $cdata_spec ) {
 				return [ 'code' => self::MANDATORY_CDATA_MISSING_OR_INCORRECT ];
 			}
 		}
+
+		// When the CDATA is expected to be JSON, ensure it's valid JSON.
+		if ( 'script' === $element->tagName && in_array( $element->getAttribute( 'type' ), [ 'application/json', 'application/ld+json' ], true ) ) {
+			if ( empty( $element->textContent ) ) {
+				return [
```

Comment:
It looks like in PHP 5.6 and earlier, `json_last_error()` returns `JSON_ERROR_NONE` if passed an empty string, for example.

In the [Changelog](https://www.php.net/manual/en/function.json-decode.php#refsect1-function.json-decode-changelog):
> 7.0.0 | An empty PHP string or value that after casting to string is an empty string (NULL, FALSE) results in JSON syntax error.

These [failed unit tests](https://travis-ci.org/ampproject/amp-wp/jobs/656510177#L1278) are from that different behavior, I think.